### PR TITLE
add reserve_ip

### PR DESCRIPTION
**Pull request by JS00000**
This PR is copied from [here](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/pull/268), which can help the socket in the UR's Teach Pendant know the host IP address when this library is running in the Docker container (especially in Docker for Windows).

### CODE REVIEW

**Comment by gokul-gokz** [comment]
I tested this PR on a ROS - melodic Docker image. Even-though, I was able to build the repository , I am unable to control the robot through the external control. After launching the driver, the following info was displayed on the terminal 
`Robot requested program.
Sent program to robot.
Robot ready to receive control commands.`
But when I try to set commands through moveit, the UR5 robot doesn't move and no response from the driver. Am i missing something?

**Comment by JS00000** [comment]
> I tested this PR on a ROS - melodic Docker image. Even-though, I was able to build the repository , I am unable to control the robot through the external control. After launching the driver, the following info was displayed on the terminal
> `Robot requested program. Sent program to robot. Robot ready to receive control commands.`
> But when I try to set commands through moveit, the UR5 robot doesn't move and no response from the driver. Am i missing something?

This PR need to build together with [this PR](https://github.com/UniversalRobots/Universal_Robots_Client_Library/pull/52). 
And when I was testing, moveit may fail to compute the trajectiry. Moveit run in Docker for Windows has some unknown issue. 
The specific reason needs more details.

**Comment by fmauch** [comment]
https://github.com/UniversalRobots/Universal_Robots_Client_Library/pull/52 merged and binary release triggered. Waiting for binary release (melodic and noetic) before merging

File: ur_robot_driver/src/hardware_interface.cpp [hunk 1]
```diff
@@ -84,6 +84,10 @@ bool HardwareInterface::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw
     return false;
   }
 
+  // IP that will be used for the robot controller to communicate back to the driver.
+  std::string reverse_ip;
+  robot_hw_nh.param<std::string>("reverse_ip", reverse_ip, "");
```
Comment (fmauch) [inline ur_robot_driver/src/hardware_interface.cpp:89]:
```suggestion
  std::string reverse_ip  robot_hw_nh.param<std::string>("reverse_ip", "");
```
This scheme is used everywhere around

File: ur_robot_driver/src/hardware_interface.cpp [hunk 2]
```diff
@@ -262,7 +266,7 @@ bool HardwareInterface::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw
         new urcl::UrDriver(robot_ip_, script_filename, output_recipe_filename, input_recipe_filename,
                            std::bind(&HardwareInterface::handleRobotProgramState, this, std::placeholders::_1),
                            headless_mode, std::move(tool_comm_setup), calibration_checksum, (uint32_t)reverse_port,
-                           (uint32_t)script_sender_port, servoj_gain, servoj_lookahead_time, non_blocking_read_));
+                           (uint32_t)script_sender_port, servoj_gain, servoj_lookahead_time, non_blocking_read_, reverse_ip));
```
Comment (fmauch) [inline ur_robot_driver/src/hardware_interface.cpp:269]: This line (this whole block) is not formatted properly. Please use clang-format to format this according to the style guidelines.

See https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/pull/350/checks?check_run_id=2762847292#step:3:511 for change required

**Review by fmauch — CHANGES_REQUESTED** [review]
Please see my comments below.

**Comment by fmauch** [comment]
Closing in favor of #412

**Pull request closed without merging**